platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/Migratable.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration;
+
+import java.nio.file.Path;
+
+/**
+ * This interface provides the mechanism for implementers to define how their data shall be
+ * exported for later import into a new system.
+ * 
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ */
+public interface Migratable {
+
+    /**
+    * Export all migratable data to the specified directory
+    *
+    * @param exportPath The exportPath is is unique path given to the {@link Migratable} 
+    *                   so it can export data without name collisions.
+    *
+    * @return Metadata describing the results of the migration
+    */
+    public MigrationMetadata export(Path exportPath) throws MigrationException;
+    
+    /**
+    * Get a description of the migratable data
+    * @return A short description of the migratable data
+    */
+    public String getDescription();
+    
+    /**
+    * Get the ID of this {@link Migratable}
+    * @return The ID of this {@link Migratable}
+    */
+    public String getId();
```

Comment:
Do we need a `getId()` method or can't we just use the service's PID? Requesting an ID is tricky because then we need to consider what format it should have, what to do if there's a conflict, etc.